Intermittent CI failure: "AssertionError: Fee of 0.00011000 BTC too low!" when running wallet_basic.py

Intermittent CI failure: "AssertionError: Fee of 0.00011000 BTC too low!" when running `wallet_basic.py`.

See [Travis log](https://travis-ci.org/github/bitcoin/bitcoin/jobs/703334035) for details.

```
…
2020-06-29T22:58:08.433000Z TestFramework (INFO): Initializing test directory /home/travis/build/bitcoin/bitcoin/ci/scratch/test_runner/test_runner_₿_🏃_20200629_225700/wallet_basic_150
2020-06-29T22:58:10.175000Z TestFramework (INFO): Mining blocks...
2020-06-29T22:58:11.456000Z TestFramework (INFO): test gettxout
2020-06-29T22:58:11.751000Z TestFramework (INFO): test gettxout (second part)
2020-06-29T22:58:13.291000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/travis/build/bitcoin/bitcoin/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 117, in main
    self.run_test()
  File "/home/travis/build/bitcoin/bitcoin/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/wallet_basic.py", line 204, in run_test
    node_0_bal = self.check_fee_amount(self.nodes[0].getbalance(), Decimal('20'), fee_per_byte, self.get_vsize(self.nodes[2].gettransaction(txid)['hex']))
  File "/home/travis/build/bitcoin/bitcoin/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/wallet_basic.py", line 44, in check_fee_amount
    assert_fee_amount(fee, tx_size, fee_per_byte * 1000)
  File "/home/travis/build/bitcoin/bitcoin/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 41, in assert_fee_amount
    raise AssertionError("Fee of %s BTC too low! (Should be %s BTC)" % (str(fee), str(target_fee)))
AssertionError: Fee of 0.00011000 BTC too low! (Should be 0.00019100 BTC)
```

https://api.cirrus-ci.com/v1/task/4686259315539968/logs/ci.log
```
11/220 - wallet_basic.py --legacy-wallet failed, Duration: 36 s
stdout:
2021-09-28T15:37:33.475000Z TestFramework (INFO): Initializing test directory /tmp/cirrus-build/ci/scratch/test_runner/test_runner_₿_🏃_20210928_153456/wallet_basic_207
2021-09-28T15:37:34.708000Z TestFramework (INFO): Mining blocks...
2021-09-28T15:37:36.232000Z TestFramework (INFO): Test gettxout
2021-09-28T15:37:36.275000Z TestFramework (INFO): Test gettxout (second part)
2021-09-28T15:37:42.855000Z TestFramework (INFO): Test sendmany
2021-09-28T15:37:44.992000Z TestFramework (INFO): Test sendmany with fee_rate param (explicit fee rate in sat/vB)
2021-09-28T15:37:45.108000Z TestFramework (INFO): Test sendmany raises 'fee rate too low' if fee_rate of 0.99999999 is passed
2021-09-28T15:37:45.115000Z TestFramework (INFO): Test sendmany raises if an invalid fee_rate is passed
2021-09-28T15:37:45.158000Z TestFramework (INFO): Test sendmany raises if an invalid conf_target or estimate_mode is passed
2021-09-28T15:37:51.913000Z TestFramework (INFO): Test -walletbroadcast
2021-09-28T15:38:02.692000Z TestFramework (INFO): Test sendtoaddress with fee_rate param (explicit fee rate in sat/vB)
2021-09-28T15:38:08.916000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 131, in main
    self.run_test()
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/wallet_basic.py", line 500, in run_test
    assert_fee_amount(fee, tx_size, Decimal(fee_rate_btc_kvb))
  File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 41, in assert_fee_amount
    raise AssertionError("Fee of %s BTC too low! (Should be %s BTC)" % (str(fee), str(target_fee)))
AssertionError: Fee of 0.00000255 BTC too low! (Should be 0.00000256 BTC)
```

--- a/test/functional/test_framework/util.py
+++ b/test/functional/test_framework/util.py
@@ -34,13 +34,14 @@ def assert_approx(v, vexp, vspan=0.00001):
         raise AssertionError("%s > [%s..%s]" % (str(v), str(vexp - vspan), str(vexp + vspan)))
 
 
-def assert_fee_amount(fee, tx_size, fee_per_kB):
-    """Assert the fee was in range"""
-    target_fee = round(tx_size * fee_per_kB / 1000, 8)
+def assert_fee_amount(fee, tx_size, feerate_BTC_kvB):
+    """Assert the fee is in range."""
+    feerate_BTC_vB = feerate_BTC_kvB / 1000
+    target_fee = satoshi_round(tx_size * feerate_BTC_vB)
     if fee < target_fee:
         raise AssertionError("Fee of %s BTC too low! (Should be %s BTC)" % (str(fee), str(target_fee)))
     # allow the wallet's estimation to be at most 2 bytes off
-    if fee > (tx_size + 2) * fee_per_kB / 1000:
+    if fee > (tx_size + 2) * feerate_BTC_vB:
         raise AssertionError("Fee of %s BTC too high! (Should be %s BTC)" % (str(fee), str(target_fee)))
 
 

